### PR TITLE
feat(ledger): check token signatures on startup

### PR DESCRIPTION
**Pull request by r4mmer**
### Acceptance Criteria
- Verify token signatures on startup

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -67,12 +68,23 @@ class Root extends React.Component {
           this.props.history.push('/wallet_type/');
         }
       });
+
+      IPC_RENDERER.on("ledger:manyTokenSignatureValid", (event, arg) => {
```
Comment (andreabadesso) [inline src/App.js:72]: Use single quotes instead of double quotes

**Review by andreabadesso — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (r4mmer) [reply src/App.js]: Done

**Review by andreabadesso — APPROVED** [review]

File: public/ledger.js [hunk 2]
```diff
@@ -424,6 +424,32 @@ class Ledger {
     }
   }
 
+  /**
+   * Verify a token and signature with ledger
```
Comment (pedroferreira1) [inline public/ledger.js:428]: This is from the other method, right?

File: public/ledger.js [hunk 3]
```diff
@@ -424,6 +424,32 @@ class Ledger {
...
+   *
+   * @param {Object} tokens Array of data to send to Ledger
+   *   version + uid + len(symbol) + symbol + len(name) + name + len(signature) + signature
+   *
+   * @return {Promise} Promise resolved when signature is received
```
Comment (pedroferreira1) [inline public/ledger.js:433]: Only when all signatures are received, right? And it resolves with a list of the failed ones, right?

File: src/utils/ledger.js [hunk 4]
```diff
@@ -203,6 +203,21 @@ if (IPC_RENDERER) {
       IPC_RENDERER.send("ledger:verifyTokenSignature", data);
     },
 
+    /**
+     * Verify token signature matches the token info on ledger
```
Comment (pedroferreira1) [inline src/utils/ledger.js:207]: This one verify many token signatures, right?

File: src/utils/wallet.js [hunk 5]
```diff
@@ -418,6 +419,26 @@ const wallet = {
     promise.then(() => {
       this.afterLoadAddressHistory();
     });
+
+    const tokenSignatures = storage.getItem('wallet:token:signatures');
```
Comment (pedroferreira1) [inline src/utils/wallet.js:423]: Should we use `tokens.getTokenSignatures`?

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**